File: WoofAndroid/design-system/src/main/kotlin/com/perrystreet/woof/designsystem/atomic/atoms/placeholder/roles/TagPlaceholderRole.kt
package com.perrystreet.woof.designsystem.atomic.atoms.placeholder.roles

import androidx.compose.runtime.Composable
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.unit.Dp
import com.perrystreet.woof.designsystem.atomic._tokens.spacing.PaddingRoles
import com.perrystreet.woof.designsystem.theme.Theme

enum class TagPlaceholderRole(val sampleText: String) {
    Expanded(sampleText = "Placeholder"),
    Regular(sampleText = "Sample tag"),
    Compact(sampleText = "Tag"),
    ;

    @Composable
    fun textStyle(): TextStyle = Theme.typography.subhead.p3

    @Composable
    fun radius(): Dp = Theme.radius.xl

    fun horizontalPadding(): PaddingRoles.Element = PaddingRoles.Element.Relaxed

    fun verticalPadding(): PaddingRoles.Element = PaddingRoles.Element.Compact
}

File: WoofIOS/SwiftPackages/DesignSystem/Sources/DesignSystem/Atomic/Atoms/Placeholder/Roles/TagPlaceholderRole.swift
import SwiftUI

public enum TagPlaceholderRole {
    case expanded
    case regular
    case compact

    public var sampleText: String {
        switch self {
        case .expanded: "Placeholder"
        case .regular: "Sample tag"
        case .compact: "Tag"
        }
    }

    public var horizontalPadding: PaddingRoles.Element { .relaxed }

    public var verticalPadding: PaddingRoles.Element { .compact }

    public func font(from theme: ThemeImplementing) -> Font {
        theme.typography.subhead.p3
    }

    public func radius(from theme: ThemeImplementing) -> CGFloat {
        theme.radius.xl
    }
}
